### ISSUE TITLE
[suggestion] Nicer `caseof` syntax

### ISSUE THREAD

**Issue by DanweDE** [geigerzaehler/sum-types#1]
A thought: How about instead of:

``` js
caseof(value, [
  [C, C2, (c) => c],
  [D, (d) => null]
])
```

also (or exclusively) allowing a imho much nicer:

``` js
caseof(value,
  C, C2,
  (c) => c,
  D,
  (d) => null
)
```

**Comment by geigerzaehler** [comment]
I like the first version. Feel free to open a PR

The second one is probably hard to parse and also (in my subjective oppinion) harder to read. So we will not add it.
